espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/ActivityHelpers.kt

```diff
@@ -104,4 +104,10 @@ object ActivityHelpers {
             instrumentation.startActivitySync(intent)
         }
     }
+    fun startActivity(params: StartActivityParams) {
```

Comment:
there is no need to create a new startActivity method. Simply change the signature for the existing one

Reply:
How about this?
`

fun startActivity(params: StartActivityParams) {
        if (params.appActivity == null && params.optionalIntentArguments == null) {
            throw IllegalArgumentException("Either activity name or intent options must be set")
        }

        val instrumentation = InstrumentationRegistry.getInstrumentation()
        val intent = if (params.optionalIntentArguments == null) {
            val fullyQualifiedAppActivity = getFullyQualifiedActivityName(instrumentation, params.appPackage, params.appActivity!!)
            val defaultOptions = mapOf<String, Any>(
                    "action" to "ACTION_MAIN",
                    "flags" to "ACTIVITY_NEW_TASK",
                    "className" to fullyQualifiedAppActivity
            )
            AndroidLogger.logger.info("Starting activity '$fullyQualifiedAppActivity' " +
                    "with default options: $defaultOptions")
            makeIntent(defaultOptions)
        } else {
            AndroidLogger.logger.info("Staring activity with custom options: $intentOptions")
            makeIntent(params.optionalIntentArguments)
        }

        makeActivityOptions(params.optionalActivityArguments).let {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
                instrumentation.startActivitySync(intent, it.toBundle())
            } else {
                instrumentation.targetContext.startActivity(intent, it.toBundle())
            }
        }
    }`

Reply:
yes, you got the idea